packages/thumbnail-grid-block/src/ThumbnailGridBlock.tsx

```diff
@@ -80,9 +80,11 @@ export const ThumbnailGridBlock = ({ appBridge }: BlockProps) => {
     };
 
     useEffect(() => {
-        setBlockSettings({ items: itemsState });
+        if (isEditing) {
+            setBlockSettings({ items: itemsState });
+        }
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [itemsState]);
+    }, [itemsState, isEditing]);
```

Comment:
This will fire an api request every time the user switches to edit mode.

What we could do is introduce a variable like isMounted, and prevent this useeffect on its first run, as I suspect that is where the error occurs in view mode.